pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -468,6 +468,57 @@ func (r *MigPlan) GetRegistryService(client k8sclient.Client) (*kapi.Service, er
 	return nil, nil
 }
 
+// Get list of migrations associated with the plan.
+// Sorted by created timestamp with final migrations grouped last.
+func (r *MigPlan) ListMigrations(client k8sclient.Client) ([]*MigMigration, error) {
+	stage := []*MigMigration{}
+	final := []*MigMigration{}
+	qlist := MigMigrationList{}
+
+	// Search
+	err := client.List(
+		context.TODO(),
+		k8sclient.InNamespace(r.Namespace),
```

Comment:
This only finds MigMigrations in the same namespace as the plan. The controller doesn't require that migrations live in a particular namespace. I just created a migration in my test env in a different namespace from the plan and it ran. Also, this pulls all migrations, even those associated with other plans. I wonder if it might make sense to add a migplan label to the migration -- perhaps actually using the existing correlation label infrastructure to add a migplan: label, or if it makes sense to use a different name here, that would be fine too. Since I don't think we can use the MigPlanRef in the selection criteria, correlation labels (or something equivalent) would allow us to find all migrations associated with this migplan regardless of namespace.

Reply:
Fixed namespace issue.
Indexed the `planRef` field and using field selector.